### PR TITLE
Supporting tags

### LINKED ISSUE #606: Feature request: Sync dbt model tags with Databricks table/view tags
### Description
Dbt allow you to [tag](https://docs.getdbt.com/reference/resource-configs/tags) your models, and you can also [tag](https://learn.microsoft.com/en-us/azure/databricks/data-governance/unity-catalog/tags) your tables and views in Databricks. There should be a way to apply dbt tags to the tables and views in Databricks.

### Alternative
I have considered implementing a custom solution using:
1. Python to read the tags from the dbt models
2. [Databricks-sql-cli](https://docs.databricks.com/en/dev-tools/databricks-sql-cli.html) to read the tags from databricks
3. Then [databricks-sql-cli](https://docs.databricks.com/en/dev-tools/databricks-sql-cli.html) again to update the tags in databricks such that they are aligned with the dbt tags

### Benefits
This is useful for filtering the tables in databricks.

I would like to contribute in creating a seamless syncing process.

### PR DESCRIPTION
**Pull request by benc-db**
Resolves #606

### Description

Adds support for setting tags (for discovery, governance) on tables/views.  Unfortunately not yet supported on MV/STs, but I built the support in the same way I do on_config_change for them, so it should be trivial to add when they become supported.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Comment by benc-db** [comment]
@jackyhu-db Yunbo suggested I add you as a reviewer while Ray is out.  Apologies that dbt code is a mix of python and jinja templating that can be pretty hard to parse.  Basic idea is that we want to support setting tags on tables/views/incremental tables.  Today we can only set tags with alter statements.  For table and views, since they are replaced on every run, we simply SET the tags the user set in databricks_tags (since dbt has its own idea of tags).  For incremental, if we have an existing table, we UNSET any tags that are present on the table but not in the config, and SET all tags that are present in the config.

**Review by jackyhu-db — APPROVED** [review]
